backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/pricing/scalePrices.feature

```diff
@@ -0,0 +1,107 @@
+      | C_OrderLine_ID.Identifier |
+      | ol_1                      |
+      | ol_2                      |
+      | ol_3                      |
+    Then validate C_OrderLine:
+      | C_OrderLine_ID.Identifier | C_Order_ID.Identifier | dateordered | M_Product_ID.Identifier | qtyordered | qtydelivered | qtyinvoiced | price | discount | currencyCode | processed |
+      | ol_1                      | o_1                   | 2022-05-17  | salesProduct_180520225  | 8          | 0            | 0           | 6     | 0        | EUR          | false     |
+      | ol_2                      | o_1                   | 2022-05-17  | salesProduct_180520225  | 15         | 0            | 0           | 6     | 0        | EUR          | false     |
+      | ol_3                      | o_1                   | 2022-05-17  | salesProduct_180520225  | 150        | 0            | 0           | 6     | 0        | EUR          | false     |
+
+
+  @from:cucumber
+  Scenario: scale price with 'Use scale price, fallback product price' set on product price
+    Given metasfresh contains M_ProductPrices
+      | Identifier | M_PriceList_Version_ID.Identifier | M_Product_ID.Identifier | PriceStd | C_UOM_ID.X12DE355 | C_TaxCategory_ID.InternalName | OPT.UseScalePrice |
+      | pp_1       | defaultPLV                        | salesProduct_180520225  | 5.00     | PCE               | Normal                        | Y                 |
+    And metasfresh contains C_Orders:
+      | Identifier | IsSOTrx | C_BPartner_ID.Identifier | DateOrdered |
+      | o_1        | true    | bpartner_1               | 2022-05-17  |
+    And metasfresh contains C_OrderLines:
+      | Identifier | C_Order_ID.Identifier | M_Product_ID.Identifier | QtyEntered |
+      | ol_1       | o_1                   | salesProduct_180520225  | 8          |
+      | ol_2       | o_1                   | salesProduct_180520225  | 15         |
+      | ol_3       | o_1                   | salesProduct_180520225  | 150        |
+    When update prices for order line
+      | C_OrderLine_ID.Identifier |
+      | ol_1                      |
+      | ol_2                      |
+      | ol_3                      |
+    And validate C_OrderLine:
+      | C_OrderLine_ID.Identifier | C_Order_ID.Identifier | dateordered | M_Product_ID.Identifier | qtyordered | qtydelivered | qtyinvoiced | price | discount | currencyCode | processed |
+      | ol_1                      | o_1                   | 2022-05-17  | salesProduct_180520225  | 8          | 0            | 0           | 5     | 0        | EUR          | false     |
+      | ol_2                      | o_1                   | 2022-05-17  | salesProduct_180520225  | 15         | 0            | 0           | 5     | 0        | EUR          | false     |
+      | ol_3                      | o_1                   | 2022-05-17  | salesProduct_180520225  | 150        | 0            | 0           | 5     | 0        | EUR          | false     |
```

Comment:
i think this feature file is a good start, but I can't see the connection between it and the moving parts of HUPricing.java where the actual fix took place.
Afais the order-lines under test don't even have an attribute or HUPI after all..

Reply:
> Afais the order-lines under test don't even have an attribute or HUPI after all..

yep, but that's what was actually broken with the HUPricing, namely if the order line didn't have any HUPI set, the HUPricing was looking at all product prices for that product

Reply:
ok then it's fine with me